guide/interactions/modals.md

```diff
@@ -0,0 +1,117 @@
+# Modals
+
+With modal interactions you can create pop-out forms that allow users to provide you with formatted input. We'll cover how to create, send and receive modals using discord.js!
+
+::: tip
+This page is a follow-up to the [interactions (slash commands) page](/interactions/slash-commands.md). Please carefully read those first so that you can understand the methods used in this section.
+:::
+
+# Building and Responding with Modals
+
+Unlike message components, modals aren't strictly components themselves. They're a callback structure used to respond to interactions.
+
+::: tip
+You can have a maximum of five `ActionRow`s per modal, and one `TextInputComponent` within an `ActionRow`. Currently, you cannot use `SelectMenuComponent`s or `ButtonComponents` in modal action rows.
+:::
+
+To create a modal you construct a new `Modal`. You can then use the setters to add the title.
+
+```js {1,7-13}
+const { Modal } = require('discord.js');
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const modal = new Modal()
+			.setCustomId('myModal')
+			.setTitle('My Modal');
+
+		// TODO: Add components to modal...
+
+		await interaction.showModal(modal);
+	}
+});
+```
+::: tip
+The custom ID is a developer-defined string of up to 100 characters.
+:::
+
+As you can see we construct the modal and assign it a custom id and a title. Once our modal is constructed we send as a response to the interaction via `#showModal`. 
+
+We're still missing one step - adding inputs. Adding inputs is similar to adding components to messages.
+
+```js {1,12-34}
+const { ActionRow, Modal, TextInputComponent, TextInputStyle } = require('discord.js');
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		// Create our modal
+		const modal = new Modal()
+			.setCustomId('myModal')
+			.setTitle('My Modal');
+
+		// Add components to modal
+
+		// Let's create our text inputs
+		const favoriteColorInput = new TextInputComponent()
+			.setCustomId('favoriteColorInput')
+		    // The label is the prompt the user sees for this input
+			.setLabel("What's your favorite color?")
+		    // Short means only a single line of text
+			.setStyle(TextInputStyle.Short);
+
+		const hobbiesInput = new TextInputComponent()
+			.setCustomId('hobbiesInput')
+			.setLabel("What's your favorites hobbies?")
+		    // Paragraph means multiple lines of text.
+			.setStyle(TextInputStyle.Paragraph);
+
+		// An action row only holds one text input,
+		// so we need one action row per text input.
+		const firstActionRow = new ActionRow().addComponents(favoriteColorInput);
+		const secondActionRow = new ActionRow().addComponents(hobbiesInput);
+
+		// Now we need to add our inputs into the modal
+		modal.addComponents(firstActionRow, secondActionRow);
+
+		// Show our modal
+		await interaction.showModal(modal);
```

Comment:
interaction.showModal is not a function

Reply:
It most definitely is, you most likely have a mistake somewhere in your code.